### DISCUSSION TITLE
SSG good on Localhost but Docker fails

### DISCUSSION THREAD

**Discussion by Uranbold** [vercel/next.js#17392] [Help] [answered]
Hello,

Problem: [[...slug]].js is working on Localhost and generated SSG but if we included it on Docker then [[...slug]].js not generating or populating data. Could you please help me someone. We don't have index.js or any files all files are connected through [[...slug]].js

package.json 

```
{
  "name": "golomt-web",
  "version": "2.0.0",
  "description": "",
  "scripts": {
    "build": "next build",
    "start": "next start",
    "dev": "NODE_OPTIONS='--inspect' next dev",
    "server": "node server.js",
    "chok": "next build && CHOKIDAR_USEPOLLING=true node server.js",
    "ssr": "next build && node server.js",
    "ssg": "next build && next start"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "autoprefixer": "9.8.0",
    "axios": "^0.19.2",
    "cacheable-response": "^2.1.4",
    "chart.js": "^2.9.3",
    "express": "^4.17.1",
    "google-maps-react": "^2.0.6",
    "jquery": "^3.5.1",
    "keyv": "^4.0.1",
    "lodash": "^4.17.15",
    "next": "^9.5.3",
    "ng2-currency-mask": "^9.0.2",
    "node-sass": "^4.14.1",
    "numeral": "^2.0.6",
    "react": "^16.13.1",
    "react-chartjs-2": "^2.9.0",
    "react-datepicker": "^3.0.0",
    "react-dom": "^16.12.0",
    "react-input-range": "^1.3.0",
    "react-number-format": "^4.4.1",
    "react-numeric-input": "^2.2.3",
    "react-redux": "^7.2.0",
    "react-select": "^3.1.0",
    "react-share": "^4.2.1",
    "react-svg-loader": "^3.0.3",
    "sass-loader": "^7.1.0",
    "uikit": "^3.5.4",
    "wpapi": "^1.2.1",
    "xml2js": "^0.4.23"
  }
}
```

Dockerfile

```FROM mhart/alpine-node
WORKDIR /app
COPY . .

RUN yarn install

RUN yarn cache clean

EXPOSE 3000

CMD ["yarn", "ssg"]```

Docker Run
`docker run -it -d --name next-frontend --env NODE_OPTIONS="--max-old-space-size=4096" -p 127.0.0.1:3000:3000 next-frontend`

Thanks.

**Comment by Uranbold** [comment, marked as answer]
Issue was HTTPS issue :) Resolved for following

`NODE_TLS_REJECT_UNAUTHORIZED='0' next build && NODE_TLS_REJECT_UNAUTHORIZED='0' next start`

Note this is only skipping the HTTPS and it's testing domain and main domain crt is included already...

**Comment by giri-jeedigunta** [comment]
> Hello,
> 
> Problem: [[...slug]].js is working on Localhost and generated SSG but if we included it on Docker then [[...slug]].js not generating or populating data. Could you please help me someone. We don't have index.js or any files all files are connected through [[...slug]].js
> 
> package.json
> 
> ```
> {
>   "name": "golomt-web",
>   "version": "2.0.0",
>   "description": "",
>   "scripts": {
>     "build": "next build",
>     "start": "next start",
>     "dev": "NODE_OPTIONS='--inspect' next dev",
>     "server": "node server.js",
>     "chok": "next build && CHOKIDAR_USEPOLLING=true node server.js",
>     "ssr": "next build && node server.js",
>     "ssg": "next build && next start"
>   },
>   "keywords": [],
>   "author": "",
>   "license": "ISC",
>   "dependencies": {
>     "autoprefixer": "9.8.0",
>     "axios": "^0.19.2",
>     "cacheable-response": "^2.1.4",
>     "chart.js": "^2.9.3",
>     "express": "^4.17.1",
>     "google-maps-react": "^2.0.6",
>     "jquery": "^3.5.1",
>     "keyv": "^4.0.1",
>     "lodash": "^4.17.15",
>     "next": "^9.5.3",
>     "ng2-currency-mask": "^9.0.2",
>     "node-sass": "^4.14.1",
>     "numeral": "^2.0.6",
>     "react": "^16.13.1",
>     "react-chartjs-2": "^2.9.0",
>     "react-datepicker": "^3.0.0",
>     "react-dom": "^16.12.0",
>     "react-input-range": "^1.3.0",
>     "react-number-format": "^4.4.1",
>     "react-numeric-input": "^2.2.3",
>     "react-redux": "^7.2.0",
>     "react-select": "^3.1.0",
>     "react-share": "^4.2.1",
>     "react-svg-loader": "^3.0.3",
>     "sass-loader": "^7.1.0",
>     "uikit": "^3.5.4",
>     "wpapi": "^1.2.1",
>     "xml2js": "^0.4.23"
>   }
> }
> ```
> 
> Dockerfile
> 
> ```
> WORKDIR /app
> COPY . .
> 
> RUN yarn install
> 
> RUN yarn cache clean
> 
> EXPOSE 3000
> 
> CMD ["yarn", "ssg"]```
> 
> Docker Run
> `docker run -it -d --name next-frontend --env NODE_OPTIONS="--max-old-space-size=4096" -p 127.0.0.1:3000:3000 next-frontend`
> 
> Thanks.
> ```

Hello devs, 

I'm facing similar issue in my local. All the routes work fine except for the routes that have slug. They straight away go to 404. I'm using `next v11` and running my app on local docker. Thanks !
